projects/gigya/src/cms-components/gigya-js/gigya-js.service.ts

```diff
@@ -0,0 +1,147 @@
+import { Injectable, NgZone } from '@angular/core';
+import {
+  BaseSiteService,
+  LanguageService,
+  ExternalJsFileLoader,
+  WindowRef,
+  GlobalMessageService,
+  AuthRedirectService,
+  GlobalMessageType,
+  User,
+  UserService,
+} from '@spartacus/core';
+import { GigyaConfig } from '../../config/gigya-config';
+import { BehaviorSubject } from 'rxjs';
+import { GigyaAuthService } from '../../auth/facade/gigya-auth.service';
+
+export function gigyaLogin(response) {
+  window['GigyaJSServiceReference'].zone.run(() =>
+    window['GigyaJSServiceReference'].component.onLoginEventHandler(response)
+  );
+}
+
+export function gigyaProfileUpdate(response) {
+  window['GigyaJSServiceReference'].zone.run(() =>
+    window['GigyaJSServiceReference'].component.onProfileUpdateEventHandler(
+      response
+    )
+  );
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class GigyaJsService {
+  loaded$ = new BehaviorSubject<boolean>(false);
+
+  constructor(
+    private gigyaConfig: GigyaConfig,
+    private baseSiteService: BaseSiteService,
+    private languageService: LanguageService,
+    private externalJsFileLoader: ExternalJsFileLoader,
+    private winRef: WindowRef,
+    private auth: GigyaAuthService,
+    private globalMessageService: GlobalMessageService,
+    private authRedirectService: AuthRedirectService,
+    private zone: NgZone,
+    private userService: UserService
+  ) {}
+
+  initialize(): void {
+    this.loadGigyaJavascript();
+    window['GigyaJSServiceReference'] = {
+      zone: this.zone,
+      onLoginEventHandlerGlobal: this.onLoginEventHandler(null),
+      onProfileUpdateEventHandlerGlobal: this.onProfileUpdateEventHandler(null),
+      component: this,
+    };
+  }
+
+  /**
+   * Method which loads the CDC Script
+   */
+  loadGigyaJavascript(): void {
+    this.baseSiteService.getActive().subscribe((baseSite) => {
```

Comment:
Keep this subscriptions in some variable and add OnDestroy handler where you unsubscribe them.